test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -0,0 +1,128 @@
+package kubernetes
```

Comment:
Why don't we just run the full k8s e2e suite, or run subsets of it? Otherwise it seems like we're potentially needlessly duplicating effort that has already gone on upstream.